makefile

```diff
@@ -22,6 +22,13 @@ install-tools: install
 	@echo Installing tools from tools.go
 	@cat ./tools/tools.go | grep _ | awk -F'"' '{print $$2}' | xargs -tI % go get %
 
+# Install integration testing tools
+install-test-tools:
+	@wget https://s3.us-west-2.amazonaws.com/dynamodb-local/dynamodb_local_latest.tar.gz
+	@sudo mkdir -p /usr/local/dynamodb
+	@sudo tar -xf dynamodb_local_latest.tar.gz -C /usr/local/dynamodb
+	@rm dynamodb_local_latest.tar.gz
```

Comment:
Does this need to install the firestore emulator as well?

My personal preference with integration tests would be to somehow containerise them, to simplify setup and separate them from our unit tests as well.

Without mocked unit testing it becomes hard to test plugin behaviour against unexpected errors from upstream services. I think for our unit tests we should implement similar mocking that we have in our go-sdk, and assert only on interactions between the plugins and their upstream services (e.g. correct translation of queries and construction of requests). Our method of emulating underlying behaviour from the mocks before was incorrect we should be asserting on method spys only, which we could auto generate with something like go-mock.